stl/inc/expected

```diff
@@ -685,6 +689,303 @@ public:
+        }
+    }
+
+    template <class _Fn>
+        requires is_copy_constructible_v<_Err>
+    constexpr auto transform(_Fn&& _Func) & {
+        using _Uty = remove_cv_t<invoke_result_t<_Fn, _Ty&>>;
+
+        if (_Has_value) {
+            if constexpr (is_void_v<_Uty>) {
+                _STD invoke(_STD forward<_Fn>(_Func), _Value);
+                return expected<_Uty, _Err>();
+            } else {
+                return expected<_Uty, _Err>(
+                    _Construct_expected_from_invoke_result_tag{}, _STD forward<_Fn>(_Func), _Value);
+            }
+        } else {
+            return expected<_Uty, _Err>(unexpect, _Unexpected);
+        }
+    }
+
+    template <class _Fn>
+        requires is_copy_constructible_v<_Err>
+    constexpr auto transform(_Fn&& _Func) const& {
+        using _Uty = remove_cv_t<invoke_result_t<_Fn, const _Ty&>>;
+
+        if (_Has_value) {
+            if constexpr (is_void_v<_Uty>) {
+                _STD invoke(_STD forward<_Fn>(_Func), _Value);
+                return expected<_Uty, _Err>();
+            } else {
+                return expected<_Uty, _Err>(
+                    _Construct_expected_from_invoke_result_tag{}, _STD forward<_Fn>(_Func), _Value);
+            }
+        } else {
+            return expected<_Uty, _Err>(unexpect, _Unexpected);
+        }
+    }
+
+
+    template <class _Fn>
+        requires is_move_constructible_v<_Err>
+    constexpr auto transform(_Fn&& _Func) && {
+        using _Uty = remove_cv_t<invoke_result_t<_Fn, _Ty&&>>;
+
+        if (_Has_value) {
+            if constexpr (is_void_v<_Uty>) {
+                _STD invoke(_STD forward<_Fn>(_Func), _STD move(_Value));
+                return expected<_Uty, _Err>();
+            } else {
+                return expected<_Uty, _Err>(
+                    _Construct_expected_from_invoke_result_tag{}, _STD forward<_Fn>(_Func), _STD move(_Value));
+            }
+        } else {
+            return expected<_Uty, _Err>(unexpect, _STD move(_Unexpected));
+        }
+    }
+
+    template <class _Fn>
+        requires is_move_constructible_v<_Err>
+    constexpr auto transform(_Fn&& _Func) const&& {
+        using _Uty = remove_cv_t<invoke_result_t<_Fn, const _Ty&&>>;
+
+        if (_Has_value) {
+            if constexpr (is_void_v<_Uty>) {
+                _STD invoke(_STD forward<_Fn>(_Func), _STD move(_Value));
+                return expected<_Uty, _Err>();
+            } else {
+                return expected<_Uty, _Err>(
+                    _Construct_expected_from_invoke_result_tag{}, _STD forward<_Fn>(_Func), _STD move(_Value));
+            }
+        } else {
+            return expected<_Uty, _Err>(unexpect, _STD move(_Unexpected));
+        }
+    }
+
+    template <class _Fn>
+        requires is_copy_constructible_v<_Ty>
+    constexpr auto transform_error(_Fn&& _Func) & {
+        using _Uty = remove_cv_t<invoke_result_t<_Fn, _Err&>>;
+

```

Comment:
Should we enforce the mandate that `_Uty` is a valid template argument for `unexpected` (contrary to the bogus requirement in the working draft, which I've filed an issue to fix) here? (Several occurrences.)